remappers/hostmetrics/hostmetrics.go

```diff
@@ -0,0 +1,107 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package hostmetrics
+
+import (
+	"path"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+const scopePrefix = "otelcol/hostmetricsreceiver"
+
+var scraperToElasticDataset = map[string]string{
+	"cpu":        "system.cpu",
+	"disk":       "system.diskio",
+	"filesystem": "system.filesystem",
+	"load":       "system.load",
+	"memory":     "system.memory",
+	"network":    "system.network",
+	"paging":     "system.memory",
+	"processes":  "system.process",
+	"process":    "system.process",
+}
+
+// Remapper maps the OTel hostmetrics to Elastic system metrics. These remapped
+// metrics power the curated Kibana dashboards. Each datapoint translated using
+// the remapper is tainted with a boolean attribute `otel_translated`.
+type Remapper struct {
+	logger *zap.Logger
+	cfg    config
+}
+
+// NewRemapper creates a new instance of hostmetrics remapper.
+func NewRemapper(logger *zap.Logger, opts ...Option) *Remapper {
+	return &Remapper{
+		cfg:    newConfig(opts...),
+		logger: logger,
+	}
+}
+
+// Remap remaps an OTel ScopeMetrics to a list of OTel metrics such that the
+// remapped metrics could be trivially converted into Elastic system metrics.
+// The current remapping logic assumes that each Metric in the ScopeMetric
+// will have datapoints for a single timestamp only.
+func (r *Remapper) Remap(src pmetric.ScopeMetrics, out pmetric.MetricSlice) {
```

Comment:
What is the reason for not implementing an opentelemetry-collector processor directly here?

Reply:
Do you mean the full processor interface ([ref](https://github.com/open-telemetry/opentelemetry-collector/blob/671dcbcac3073c8001836e4b90354c345fce0d69/processor/processor.go#L28-L31))? -- I didn't do this because we didn't need to. The `component.Component` felt unnecessary complexity for our use-case.

The consumer interface ([ref](https://github.com/open-telemetry/opentelemetry-collector/blob/671dcbcac3073c8001836e4b90354c345fce0d69/consumer/metrics.go#L14-L18)) was a better alternative but I decided against it because it would be too restrictive. My goal was to have an interface that can be used in a multiple ways and easily adapted to implement a consumer interface if needed - the reason why I separated the output `pmetric.MetricSlice` from the source's `pmetric.ScopeMetrics`. Let me know if it doesn't make sense, I can adopt the consumer interface.

Reply:
> It's not a big deal, and we can refactor as needed. I was wondering if there was something fundamentally preventing us - sounds like no.

Yeah, no blockers in doing that.